experimental/framework/fluid-static/src/FluidStatic.ts

```diff
@@ -41,36 +42,36 @@ export interface ContainerCreateConfig extends ContainerConfig {
 export class FluidContainer {
     private readonly types: Set<string>;
     public constructor(
-        private readonly container: IContainer,
+        container: IContainer, // we anticipate using this later, e.g. for Audience
         namedRegistryEntries: NamedFluidDataStoreRegistryEntry[],
+        private readonly rootDataObject: RootDataObject,
         public readonly createNew: boolean) {
-        this.types = new Set();
-        namedRegistryEntries.forEach((value: NamedFluidDataStoreRegistryEntry) => {
-            const type = value[0];
-            if (this.types.has(type)) {
-                throw new Error(`Multiple DataObjects share the same type identifier ${value}`);
-            }
-            this.types.add(type);
-        });
-    }
+            this.types = new Set();
+            namedRegistryEntries.forEach((value: NamedFluidDataStoreRegistryEntry) => {
+                const type = value[0];
+                if (this.types.has(type)) {
+                    throw new Error(`Multiple DataObjects share the same type identifier ${value}`);
+                }
+                this.types.add(type);
+            });
+        }
 
-    public async createDataObject<T extends DataObject>(
-        dataObjectClass: IFluidStaticDataObjectClass, id: string): Promise<T> {
+    public async createDataObject<T extends IFluidLoadable = IFluidLoadable>(
```

Comment:
I'm not convinced that IFluidLoadable is the route we want to go here but I'm okay pushing it in with this change. I think we will want the simplicity that DataObject provides but.

Reply:
Reverted in latest